### PR TITLE
lib/discover: Don't leak relay-tokens to discovery

### PR DESCRIPTION
**Pull request by entity0xfe**
This PR aims to mitigate the token-disclosure described in https://github.com/syncthing/syncthing/pull/8561#issuecomment-1312547174 by @GermanCoding: The client leaks the relay-token to discovery by announcing it as part of the `relay://...`-URL.

I tested this PR with network local and external clients (read: local and global discovery) but would appreciate it if somebody else could give it a go.

### CODE REVIEW

**Comment by calmh** [comment]
The question in my mind is whether/why we want to pass on any query parameters whatsoever to discovery...

**Comment by GermanCoding** [comment]
As I understand the current logic, we do submit the query part of the relay server URL because it contains the relay server ID. ST clients wanting to dial another device via a relay address want to open a relay session in temporary mode. In order to perform relay server certificate validation, they need to know the ID. They can get that via the announced URL as part of the query.

**Comment by calmh** [comment]
Good point, so that needs to be sent. But currently we just send the full URL as we got it from the relay pool endpoint (or manual config), which contains all kinds of stuff like who provides it etc.

**Comment by GermanCoding** [comment]
So we perhaps want to invert the logic (block vs allowlist)? Only include query parameters we explicitly want announced to discovery (the ID) and ignore everything else?

**Comment by entity0xfe** [comment]
Something like this? I think the checks failing has nothing to do with the code.

**Review by imsodin — APPROVED** [review]
This looks good - thanks for updating it!

Just one naming suggestion - already approved though.

**Comment by imsodin** [comment]
And yes, the check is definitely not on this PR - probably some FS/OS limitation of the github actions runner (those jobs are new-ish - nice btw :) ).

File: lib/discover/local.go [hunk 1]
```diff
@@ -315,3 +318,32 @@ func filterUnspecifiedLocal(addrs []string) []string {
 	}
 	return filtered
 }
+
+func sanitizeRelayAddresses(addrs []string) []string {
+	filtered := addrs[:0]
+	whitelist := []string{"id"}
```
[the remark this answers is not in the document]
Comment (entity0xfe) [reply lib/discover/local.go]: sure, done.

**Review by imsodin — APPROVED** [review]

**Pull request merged**